pkgs/tools/filesystems/snapraid/default.nix

```diff
@@ -1,19 +1,21 @@
-{ stdenv, fetchFromGitHub, autoreconfHook }:
+{ stdenv, fetchgit, autoreconfHook, git }:
 
 stdenv.mkDerivation rec {
   name = "snapraid-${version}";
-  version = "11.1";
+  version = "11.2";
 
-  src = fetchFromGitHub {
-    owner = "amadvance";
-    repo = "snapraid";
-    rev = "v${version}";
-    sha256 = "1vgyiagn6qvyyz4jkf8pzlg0w9apb6mzza97a9x03p82fav2afx0";
+  src = fetchgit {
+    url = "https://github.com/amadvance/snapraid";
+    rev = "refs/tags/v${version}";
+    sha256 = "05hq1478k61pa1hzg6xjaw63z6nba3yd7k2fxsxpb1fw06j5ywmj";
+    # needed by git for determining the version number
+    deepClone = true;
+    leaveDotGit = true;
```

Comment:
Uh, this is always ugly and often break the checksum. Is it possible to do a simple patch instead?

Reply:
You can simple say `VERSION=version;` in nix to make it a environment variable: https://github.com/amadvance/snapraid/blob/f31dfada3e52e03487d4f68cfacb3b984381385b/autover.sh#L14

Reply:
You are right. This is a much better choice.